provider/aws/aws.go

```diff
@@ -376,6 +376,13 @@ func (p *AWSProvider) records(ctx context.Context, zones map[string]*route53.Hos
 	for _, z := range zones {
 		params := &route53.ListResourceRecordSetsInput{
 			HostedZoneId: z.Id,
+			// From the experiments, it seems that the default MaxItems applied is 100,
+			// and that, on the server side, there is a hard limit of 300 elements per page.
+			// After a discussion with AWS representants, clients should accept
+			// when less items are returned, and still paginate accordingly.
+			// As we are using the standard AWS client, this should already be compliant.
+			// Hence, ifever AWS decides to raise this limit, we will automatically reduce the pressure on rate limits
+			MaxItems: aws.String("1000"),
```

Comment:
I would prefer having this as `300` as it's the limit now.

I would also love to see a constant declared for that instead of a magic ~number~ string.

Reply:
done :D